sdk/core/perf/CMakeLists.txt

```diff
@@ -42,7 +44,7 @@ endif()
 
 # make sure that users can consume the project as a library.
 add_library (Azure::Perf ALIAS azure-perf)
-target_link_libraries(azure-perf PRIVATE azure-core)
+target_link_libraries(azure-perf PUBLIC azure-core)
```

Comment:
Please help me understand what is motivating this change.

Reply:
In summary, if the library (the perf-fw) is consuming the azure-core lib from source files only (.cpp), then it can link PRIVATE. In that case, the azure-core headers are not available to the perf-fw headers but only to the src files.

In this PR, I am adding a new header to the perf-fw which provides a derived class from BodyStream, so now a header needs to include a header from azure-core, which requires a PUBLIC link.